DeployIntegrationArtifact Command (#2474)
SAP/jenkins-library: documentation/docs/steps/deployIntegrationArtifact.md

# ${docGenStepName}

## ${docGenDescription}

## Prerequisites

With this step you can deploy a integration flow artifact in to SAP Cloud Platform integration runtime using OData API.

> Learn more about the SAP Cloud Integration remote API for deploying an integration artifact [here](https://help.sap.com/viewer/368c481cd6954bdfa5d0435479fd4eaf/Cloud/en-US/08632076a1114bc1b6a1ecafef8f0178.html).

Reviewer: These are not prerequisites. This should be under the description heading above, which in turn is generated from the metadata yaml file [here](https://github.com/SAP/jenkins-library/pull/2474/files#diff-c2f5178a483ee6ac8bd9dc81d16c63d1a9d2e775d82c5c8e46bd6834f37a00bfR3)

## ${docGenParameters}

## ${docGenConfiguration}

## ${docJenkinsPluginDependencies}

## Example

Example configuration for the use in a `Jenkinsfile`.

```groovy
deployIntegrationArtifact script: this
```

> Example for the use in a YAML configuration file (such as `.pipeline/config.yaml`).

Reviewer: ```suggestion
Example of a YAML configuration file (such as `.pipeline/config.yaml`).
```

```yaml
steps:
  <...>
  deployIntegrationArtifact:
    cpiCredentialsId: 'MY_CPI_OAUTH_CREDENTIALSID_IN_JENKINS'

>     integrationFlowId: 'INTEGRATION_FLOW_NAME'
>     integrationFlowVersion: 'INTEGRATION_FLOW_VERSION'

Reviewer: How about using my here too:
```suggestion
    integrationFlowId: 'MY_INTEGRATION_FLOW_NAME'
    integrationFlowVersion: 'MY_INTEGRATION_FLOW_VERSION'
```

# ${docGenStepName}

## ${docGenDescription}

Deploy a CPI integration flow
With this step you can deploy a integration flow artifact in to SAP Cloud Platform integration runtime using OData API.

Learn more about the SAP Cloud Integration remote API for deploying an integration artifact [here](https://help.sap.com/viewer/368c481cd6954bdfa5d0435479fd4eaf/Cloud/en-US/08632076a1114bc1b6a1ecafef8f0178.html)

> 

Reviewer: This is not needed as it has moved to the long description in the metadata yaml file and thus will be generated here with the `${docGenDescription}` placeholder.

# ${docGenStepName}

> ${docGenDescription}

Reviewer: This should still be a subheading though, starting with two `##`

# ${docGenStepName}

## ${docGenDescription}
With this step you can deploy a integration flow artifact in to SAP Cloud Platform integration runtime using OData API.

> Learn more about the SAP Cloud Integration remote API for deploying an integration artifact [here](https://help.sap.com/viewer/368c481cd6954bdfa5d0435479fd4eaf/Cloud/en-US/08632076a1114bc1b6a1ecafef8f0178.html).

Reviewer: I believe the result will be the description of the metadata file followed by this text, so:
```
Deploy a CPI integration flow
With this step you can deploy a integration flow artifact in to SAP Cloud Platform integration runtime using OData API.

Learn more about the SAP Cloud Integration remote API for deploying an integration artifact [here](https://help.sap.com/viewer/368c481cd6954bdfa5d0435479fd4eaf/Cloud/en-US/08632076a1114bc1b6a1ecafef8f0178.html).
```
